pennylane/numpy/random.py

```diff
@@ -15,7 +15,44 @@
 This package provides a wrapped version of autograd.numpy.random, such that
 it works with the PennyLane :class:`~.tensor` class.
 """
+from packaging.version import parse as parse_version
+
 from autograd.numpy import random as _random
-from .wrapper import wrap_arrays
+from numpy import __version__ as np_version
+from numpy.random import MT19937, PCG64, Philox, SFC64  # pylint: disable=unused-import
+
+from .wrapper import wrap_arrays, tensor_wrapper
 
 wrap_arrays(_random.__dict__, globals())
+
+if parse_version(np_version) > parse_version("1.17.0"):
+
+    # pylint: disable=too-few-public-methods
+    # pylint: disable=missing-class-docstring
+    class Generator(_random.Generator):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+            self.__doc__ = "PennyLane wrapped NumPy Generator object\n" + super().__doc__
+
+            for name in dir(_random.Generator):
+                if name[0] != "_":
+                    self.__dict__[name] = tensor_wrapper(getattr(super(), name))

```

Comment:
Nice! I tested this out, it works really well.

Reply:
I had a look through all methods - it looks like most of them do return `ndarray` objects, so it should be good 👍 Are we inadvertently wrapping any that _don't_ return arrays?

Reply:
The methods will return a single scalar if not called with a size.  If a size is provided, then it returns an `ndarray`.  I've thought about explicitly writing out the names of all the generators that need to be wrapped, but decided against it.